Redeclaration of variable with no linkage

**V version:** V 0.1.24 778a1cc

**OS:** Linux lambda 4.19.91-1-MANJARO #1 SMP Sat Dec 21 11:18:44 UTC 2019 x86_64 GNU/Linux

**What did you do?**
```v
import os

fn main () {
	mut s := ''
	s = os.read_file("mike.dat") or {
		panic('boom')
	}
	println(s)
}
```

**What did you expect to see?**

content of file mike.dat
 
**What did you see instead?**
```
$ v run utf.v
/home/mpech/.cache/v/utf.tmp.c: In function ‘main__main’:
/home/mpech/.cache/v/utf.tmp.c:5229:10: error: redeclaration of ‘s’ with no linkage
 5229 |  string  s   ;
      |          ^
/home/mpech/.cac...
(Use `v -g` to print the entire error message)

V error: C error. This should never happen. 
Please create a GitHub issue: https://github.com/vlang/v/issues/new/choose
```

--- a/vlib/compiler/gen_c.v
+++ b/vlib/compiler/gen_c.v
@@ -116,7 +116,7 @@ fn (p mut Parser) gen_handle_option_or_else(_typ, name string, fn_call_ph int) s
 		is_mut: false
 		is_used: true
 	})
-	if is_assign && !name.contains('.') {
+	if is_assign && !name.contains('.') && !p.is_var_decl {
 		// don't initialize struct fields
 		p.genln('$typ $name;')
 	}
@@ -717,4 +717,3 @@ fn (p mut Parser) gen_array_push(ph int, typ, expr_type, tmp, elm_type string) {
 		p.gen('), $tmp, $elm_type)')
 	}
 }
-

